<?xml version="1.0"?>
<launch>
	<!-- This launch file connects the follower to the ground control station and does nothing else. -->
	<!-- LOAD FOLLOWER PARAMETER FILES (WITHIN FOLLOWER NAMESPACE) -->
	<rosparam command="load" file="param/follower_actual_params.yaml"/>
	<!-- MAVROS configs -->
	<arg name="fcu_url" default="/dev/ttyUSB0:921600"/>
	<arg name="respawn_mavros" default="false"/>
	<!-- MAVROS -->
	<include file="$(find mavros)/launch/px4.launch">
		<!-- GCS link is provided by SITL -->
		<arg name="gcs_url" value="udp://:14550@192.168.1.102"/>
		<arg name="fcu_url" value="$(arg fcu_url)"/>
		<arg name="respawn_mavros" value="$(arg respawn_mavros)"/>
	</include>
  <!-- VICON VRPN CLIENT SETUP -->
  <arg name="server" default="192.168.1.114"/>
  <remap from="/vrpn_client_node/UWBFYP_follower/pose" to="/mavros/vision_pose/pose"/>
  <node pkg="vrpn_client_ros" type="vrpn_client_node" name="vrpn_client_node" output="screen">
    <rosparam subst_value="true">
      server: $(arg server)
      port: 3883
      update_frequency: 100.0
      frame_id: world
      # Use the VRPN server's time, or the client's ROS time.
      use_server_time: false
      broadcast_tf: true
      # Must either specify refresh frequency > 0.0, or a list of trackers to create
      refresh_tracker_frequency: 1.0
    </rosparam>
  </node>

</launch>
